src/secret_sharing/replicated/mod.rs

```diff
@@ -0,0 +1,2 @@
+pub mod malicious;
+pub mod semi_honest;
```

Comment:
This is probably my one reservation about this, though I'm on the fence about it.

You have a layer here where there are no peers, which suggests that you could maybe flatten this.  That said, this is nicely descriptive.